### DISCUSSION TITLE
Open mobile app when signing message

### DISCUSSION THREAD

**Discussion by morleyzhi** [rainbow-me/rainbowkit#549] [Q&A]
Our app has the common sign-in workflow of connecting a user's wallet, then asking them to sign a message in the wallet to sign up. We're using RainbowKit to initiate the connection, then Wagmi to detect the connection and sign the message.

We noticed that on mobile, the initial connection opens the wallet app they choose (thanks to [this](https://github.com/rainbow-me/rainbowkit/blob/main/packages/rainbowkit/src/components/ConnectOptions/MobileOptions.tsx#L60-L77)), but requesting a signature does not.

What is the recommended way to make it open the app? The best strategies I can think of are:

- Patch RainbowKit to expose the wallet the user taps on before they connect, then use the `{ wallet }` export of RainbowKit to fetch the mobile URI if there is one.
- Initialiize all the `{ wallet }` instances, and check for the recent flag.
- ???

Some pseudocode:

```
import { useConnect, useSignMessage } from 'wagmi';

const SignIn = () => {
  const { isConnected } = useConnect();
  const { signMessageAsync } = useSignMessage();

  const message = new SiweMessage(); 

  const requestSignature = useCallback(async () => {
      const signature = await signMessageAsync({ message });

      // todo: window.open the wallet they connected to

      signIn('credentials', {
        message: JSON.stringify(message),
        signature,
        callbackUrl: '/',
  }, [signMessageAsync]);

  useEffect(() => {
    if (isConnected) {
      requestSignature();
    }
  }, [
    isConnected,
    requestSignature,
  ]);
  return <ConnectButton />;
```

**Comment by FranRom** [comment]
Did you find a solution for that? I'm experiencing the same
`We noticed that on mobile, the initial connection opens the wallet app they choose, but requesting a signature does not.`
